### PR TITLE
[tests/override_config_table] load_minigraph with Golden Config test

### PR DESCRIPTION
**Pull request by wen587**
### Description of PR

Summary: E2E test for load minigraph with Golden Config file
Wait sonic-utilities feature https://github.com/Azure/sonic-utilities/pull/2140 merge in sonic-buildimage
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
This PR is to verify load minigraph with Golden Config and see if Golden Config overrides minigraph.
This test contains below phases:
I. Empty Golden Config
II. Partial Golden Config
III. New Feature in Golden Config
IV. Full Golden Config
#### How did you do it?
Write E2E test to verify
#### How did you verify/test it?
Run E2E test on kvm

### CODE REVIEW

File: tests/override_config_table/test_override_config_table.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+
+def restore_config(duthost, config, config_backup):
+    logger.info("Restore {} with {} on {}".format(
+        config, config_backup, duthost.hostname))
+    duthost.shell("mv {} {}".format(config_backup, config))
+
+
+def get_running_config(duthost):
+    return json.loads(duthost.shell("sonic-cfggen -d --print-data")['stdout'])
```
Comment (qiluo-msft) [inline tests/override_config_table/test_override_config_table.py:38]: >"sonic-cfggen -d --print-data")['stdout'] [](http://example.com/codeflow?start=36&length=41)

Is it better to use `show runningconfiguration all`?

File: tests/override_config_table/test_override_config_table.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    # Reload test env with minigraph
+    config_reload(duthost, config_source="minigraph", safe_reload=True)
+    running_config = get_running_config(duthost)
+
+    yield running_config
+
+    # Restore configDB after test.
+    restore_config(duthost, CONFIG_DB, CONFIG_DB_BACKUP)
+    # Restore Golden Config after test.
+    if golden_config_exists_on_dut:
```
Comment (qiluo-msft) [inline tests/override_config_table/test_override_config_table.py:71]: >golden_config_exists_on_dut [](http://example.com/codeflow?start=7&length=27)

For `else` case, you need to remove any GOLDEN_CONFIG  file you created in your own testcases.

File: tests/override_config_table/test_override_config_table.py (hunk 1 above)
Comment (wen587) [reply tests/override_config_table/test_override_config_table.py:38]: I think they are the same.

File: tests/override_config_table/test_override_config_table.py (hunk 2 above)
Comment (wen587) [reply tests/override_config_table/test_override_config_table.py:71]: Thanks for pointing it out.

**Review by qiluo-msft — DISMISSED** [review]

File: tests/override_config_table/test_override_config_table.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+    # Restore configDB after test.
+    restore_config(duthost, CONFIG_DB, CONFIG_DB_BACKUP)
+    # Restore Golden Config after test, else cleanup test file.
+    if golden_config_exists_on_dut:
+        restore_config(duthost, GOLDEN_CONFIG, GOLDEN_CONFIG_BACKUP)
+    else:
+        duthost.file(path=GOLDEN_CONFIG, state='absent')
+
+    # Restore config before test
```
Comment (qiluo-msft) [inline tests/override_config_table/test_override_config_table.py:76]: >before [](http://example.com/codeflow?start=21&length=6)

You mean "after this test"?

Comment (wen587) [reply tests/override_config_table/test_override_config_table.py:76]: Mean restore to the config when the test is not started.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**